from PyQt5 import QtCore, QtWidgets
from bd import all_results, all_results2, all_results3, all_results4


class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(1063, 587)
        MainWindow.setStyleSheet("background-color: rgb(199, 199, 199);")
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.centralwidget)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.verticalLayout_2 = QtWidgets.QVBoxLayout()
        self.verticalLayout_2.setObjectName("verticalLayout_2")
        self.verticalLayout = QtWidgets.QVBoxLayout()
        self.verticalLayout.setObjectName("verticalLayout")
        self.comboBox = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox.setMaximumSize(QtCore.QSize(501, 16777215))
        self.comboBox.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.comboBox.setObjectName("comboBox")
        self.verticalLayout.addWidget(self.comboBox)
        self.comboBox_3 = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox_3.setMaximumSize(QtCore.QSize(501, 16777215))
        self.comboBox_3.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.comboBox_3.setObjectName("comboBox_3")
        self.verticalLayout.addWidget(self.comboBox_3)
        self.comboBox_2 = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox_2.setMaximumSize(QtCore.QSize(501, 16777215))
        self.comboBox_2.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.comboBox_2.setObjectName("comboBox_2")
        self.verticalLayout.addWidget(self.comboBox_2)
        self.comboBox_4 = QtWidgets.QComboBox(self.centralwidget)
        self.comboBox_4.setMaximumSize(QtCore.QSize(501, 16777215))
        self.comboBox_4.setStyleSheet("background-color: rgb(255, 255, 255);")
        self.comboBox_4.setObjectName("comboBox_4")
        self.comboBox_2.setStyleSheet('selection-background-color: rgb(150,150,150)')
        self.comboBox_3.setStyleSheet('selection-background-color: rgb(150,150,150)')
        self.comboBox_4.setStyleSheet('selection-background-color: rgb(150,150,150)')
        self.comboBox.setStyleSheet('selection-background-color: rgb(150,150,150)')
        self.verticalLayout.addWidget(self.comboBox_4)
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setMaximumSize(QtCore.QSize(501, 16777215))
        self.label_2.setStyleSheet("font: 75 12pt \"Calibri\";")
        self.label_2.setObjectName("label_2")
        self.verticalLayout.addWidget(self.label_2)
        self.doubleSpinBox = QtWidgets.QDoubleSpinBox(self.centralwidget)
        self.doubleSpinBox.setMaximumSize(QtCore.QSize(501, 16777215))
        self.doubleSpinBox.setStyleSheet("background-color: rgb(255, 170, 0);")
        self.doubleSpinBox.setObjectName("doubleSpinBox")
        self.verticalLayout.addWidget(self.doubleSpinBox)
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setMaximumSize(QtCore.QSize(501, 16777215))
        self.label.setStyleSheet("font: 75 12pt \"Calibri\";")
        self.label.setObjectName("label")
        self.verticalLayout.addWidget(self.label)
        self.doubleSpinBox_2 = QtWidgets.QDoubleSpinBox(self.centralwidget)
        self.doubleSpinBox_2.setMaximumSize(QtCore.QSize(501, 16777215))
        self.doubleSpinBox_2.setStyleSheet("background-color: rgb(170, 255, 255);")
        self.doubleSpinBox_2.setObjectName("doubleSpinBox_2")
        self.verticalLayout.addWidget(self.doubleSpinBox_2)
        self.verticalLayout_2.addLayout(self.verticalLayout)
        self.pushButton = QtWidgets.QPushButton(self.centralwidget)
        self.pushButton.setMaximumSize(QtCore.QSize(503, 16777215))
        self.pushButton.setStyleSheet("background-color: rgb(255, 255, 255);\n"
                                      "selection-background-color: rgb(0, 85, 255);")
        self.pushButton.setObjectName("pushButton")
        self.verticalLayout_2.addWidget(self.pushButton)
        self.listWidget = QtWidgets.QListWidget(self.centralwidget)
        self.listWidget.setMaximumSize(QtCore.QSize(501, 16777215))
        self.listWidget.setObjectName("listWidget")
        self.verticalLayout_2.addWidget(self.listWidget)
        self.horizontalLayout.addLayout(self.verticalLayout_2)
        self.GraphWidget = PlotWidget(self.centralwidget)
        self.GraphWidget.setObjectName("GraphWidget")
        self.horizontalLayout.addWidget(self.GraphWidget)
        MainWindow.setCentralWidget(self.centralwidget)
        self.doubleSpinBox.setMaximum(1000)
        self.doubleSpinBox_2.setMaximum(1000)
        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        for i in range(len(all_results)):
            self.comboBox.addItem(all_results[i][1])
        for i in range(len(all_results2)):
            self.comboBox_2.addItem(all_results2[i][1])
        for i in range(len(all_results3)):
            self.comboBox_4.addItem(all_results3[i][1])
        for i in range(len(all_results4)):
            self.comboBox_3.addItem(all_results4[i][1])

        self.label_2.setText(_translate("MainWindow", "Температура"))
        self.label.setText(_translate("MainWindow", "Напряжение"))
        self.pushButton.setText(_translate("MainWindow", "Вывести результат"))


from pyqtgraph import PlotWidget
